ansible/roles/csm.sbps.dns_srv_records/files/sbps_dns_srv_records.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
+#
+# MIT License
+#
+# (C) Copyright 2024 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+set -euo pipefail
+
+PDNS_API_KEY=$(kubectl -n services get secret cray-powerdns-credentials -o jsonpath='{.data.pdns_api_key}' | base64 -d)
+PDNS_API=$(kubectl -n services get svc cray-dns-powerdns-api -o jsonpath='{.spec.clusterIP}')
+
+hsn_srv_records=""
+nmn_srv_records=""
+hsn_a_records=""
+nmn_a_records=""
+
+system_name="$(cat /etc/cray/system_name)"
+
+while read -r line; do
```

Comment:
How/where does this script get called?  What is the format of the data that is being passed to it on standard input?

Reply:
From _"ansible/roles/csm.sbps.dns_srv_records/tasks/main.yml"_ play book,
1. the first task invokes the script _"sbps_get_hsn_nmn.sh"_ to run on each worker node (_hosts: Management_Worker_ as defined in top level _"config_sbps_iscsi_targets.yml"_ for this role). This script gets host name, HSN IP and NMN IP of the worker node and the format that it echo to stdout is: _"host_name:hsn_ip:nmn_ip"_ (Ex: _ncn-w001:10.253.0.2:10.252.1.12_). 
2. then the second task updates (appends) above stdout output for each of the worker node as it runs as a record to "/tmp/hsn_nmn_info.txt"
3. finally the 3rd task invokes _"sbps_dns_srv_records.sh"_ with each record as input from "/tmp/hsn_nmn_info.txt" and creates DNS "SRV" and "A" records for all the worker nodes dynamically and finally updates these records as data with curl PATCH (update) from local host (_delegate_to: localhost_).

Reply:
I have also added/ updated comments in the files wherever necessary.